### PR TITLE
Mention file:line in Already defined as ... error message

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/test/elixir/kernel/errors_test.exs [hunk 1]
```diff
@@ -888,12 +888,14 @@ defmodule Kernel.ErrorsTest do
...
 
   test "def defmacro clause change" do
-    assert_eval_raise CompileError, "nofile:3: defmacro foo/1 already defined as def", '''
-    defmodule Kernel.ErrorsTest.DefDefmacroClauseChange do
-      def foo(1), do: 1
-      defmacro foo(x), do: x
-    end
-    '''
+    assert_eval_raise CompileError,
+                      "nofile:3: defmacro foo/1 already defined as def in nofile:3",
```
Comment (josevalim) [inline lib/elixir/test/elixir/kernel/errors_test.exs:892]: You are using the wrong File:Line, given it is just being repeated in the error message. You would have to pick the other File:Line pairing.

Comment (eksperimental) [reply lib/elixir/test/elixir/kernel/errors_test.exs]: You are right.  Fixed!

File: lib/elixir/src/elixir_def.erl [hunk 2]
```diff
@@ -246,7 +246,7 @@ store_definition(Check, Kind, Meta, Name, Arity, File, Module, Defaults, Clauses
   {MaxDefaults, FirstMeta} =
     case ets:lookup(Set, {def, Tuple}) of
       [{_, StoredKind, StoredMeta, StoredFile, StoredCheck, {StoredDefaults, LastHasBody, LastDefaults}}] ->
-        check_valid_kind(Meta, File, Name, Arity, Kind, StoredKind),
+        check_valid_kind(Meta, File, Name, Arity, Kind, StoredKind, StoredMeta),
```
Comment (josevalim) [inline lib/elixir/src/elixir_def.erl:249]: You need to pass StoredFile too and use that in the error message too.

Comment (eksperimental) [reply lib/elixir/src/elixir_def.erl:249]: Sorry. Fixed

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**